docs/datatypes/concurrency/ref.md

```diff
@@ -52,17 +48,17 @@ val stateRef = Ref.make(Active)
 
 > _**Warning**_:  
 >
-> The big mistake to creating `Ref` is trying to store mutable data inside it. It doesn't work. The only way to use a `Ref` is to store **immutable data** inside it, otherwise, it does not provide us atomic guarantees, and we can have collisions and race conditions. 
+> The big mistake to creating a `Ref` is trying to store mutable data inside it, which doesn't work. The only way to use a `Ref` is to store **immutable data** inside it. We are otherwise not provided atomic guarantees, which means we can have collisions and race conditions. 
```

Comment:
```suggestion
> The big mistake when creating a `Ref` is trying to store mutable data inside it, which doesn't work. A`Ref` must be used with **immutable data**. Otherwise, we lose our atomic guarantees, which can lead to collisions and race conditions. 
```